qp_klp/MetagenomicStep.py

```diff
@@ -0,0 +1,159 @@
+from os import walk
+from os.path import exists, join, basename
+from qp_klp.klp_util import FailedSamplesRecord
+from sequence_processing_pipeline.PipelineError import PipelineError
+import pandas as pd
+from qp_klp.Step import Step
+
+
+class MetagenomicStep(Step):
+    def __init__(self, pipeline, master_qiita_job_id, sn_tid_map_by_project,
+                 status_update_callback=None):
+        super().__init__(pipeline,
+                         master_qiita_job_id,
+                         sn_tid_map_by_project,
+                         status_update_callback)
+
+        if pipeline.pipeline_type != 'metagenomic':
+            raise ValueError("Cannot instantiate MetagenomicStep object from "
+                             f"pipeline of type '{pipeline.pipeline_type}'")
+
+        # Note: FailedSamplesRecord is not used when processing amplicon as the
+        # samples are processed as a single fastq file and hence that info
+        # is not available.
+        self.fsr = FailedSamplesRecord(self.pipeline.output_path,
+                                       pipeline.sample_sheet.samples)
+        self.project_names = None
+
+    def convert_bcl_to_fastq(self):
```

Comment:
Yup, this looks like duplicated code between the two classes.

Reply:
Going to skip to the next section.

Reply:
In the case of convert_bcl_to_fastq(), one uses configuration info for bcl2fastq while the other uses bcl-convert. Also the metagenomic version generates a failed sample report afterward. Not sure where you're seeing the duplication.